pkg/multi-tenancy/read/read.go

```diff
@@ -0,0 +1,57 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package read
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/timescale/promscale/pkg/multi-tenancy/config"
+)
+
+type readAuthorizer struct {
+	config.Config
+	// mtSafetyLabelPair is a label-pair that is applied to incoming multi-tenant read requests for security reasons.
+	// This matcher helps prevent a query from querying a tenant for the query has not been authorized.
+	mtSafetyLabelMatcher *labels.Matcher
+}
+
+// NewPlainReadAuthorizer is a authorizer that does not care about the token checks. It just makes sure that the
+// read operations are performed against a valid tenants, which Promscale has been asked for.
+func NewAuthorizer(cfg config.Config) (Authorizer, error) {
+	if _, ok := cfg.(*config.OpenConfig); ok {
```

Comment:
maybe the config interface should have an AllowAllTenants() function instead?

Reply:
Yes, that would be better